### PR TITLE
[#157539460] Fix api specs

### PR DESCRIPTION
**Pull request by lussoluca**
To correctly return the right responses to the client I've introduced the same `IResponse` interface used in digital-citizenship-functions, and then I've adapted the rest of the code and the test.

### CODE REVIEW

**Comment by cloudify** [comment]
awesome

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Remember to fix the PR title by removing WIP wording when ready
  </td>
    </tr>
  </tbody>
</table>



## Affected stories

  * 🐞 [#157539460](https://www.pivotaltracker.com/story/show/157539460): Le spec dell'API del backend sono incomplete

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/app.ts [hunk 1]
```diff
@@ -190,7 +160,8 @@ export function newApp(
   );
 
   app.get("/metadata", (_, res: express.Response) => {
-    acsController.metadata(res);
+    const response = acsController.metadata();
```
Comment (cloudify) [inline src/app.ts]: is this sync?

File: src/app.ts [hunk 2]
```diff
@@ -203,40 +174,45 @@ export function newApp(
...
+      response.apply(res);
     }
   );
 
   app.get(
     "/api/v1/services/:id",
     bearerTokenAuth,
-    (req: express.Request, res: express.Response) => {
-      servicesController.getService(req, res);
+    async (req: express.Request, res: express.Response) => {
```
Comment (cloudify) [inline src/app.ts]: do I see a pattern here? :D

```ts
function toExpressHandler<T>(handler: (req: express.Request) => Promise<IResponse<T>>): async (req: express.Request, res: express.Response) => void {
  return async (req, res) => {
    const response = await handler(req);
    response.apply(res);
  }
}
```

**Comment by gunzip** [comment]
that's right, I can do that (is there anyone working on this @lussoluca @cloudify ?). There a couple of responses that cannot be moved due to some documentdb dep. ie. https://github.com/teamdigitale/digital-citizenship-functions/blob/master/lib/utils/response.ts#L329

**Comment by cloudify** [comment]
go ahead, yes move the responses that have no deps and can be easily reused

**Comment by gunzip** [comment]
moved responses to italia-ts-commons @lussoluca

File: src/controllers/authenticationController.ts [hunk 3]
```diff
@@ -161,55 +163,47 @@ export default class AuthenticationController {
...
    * The Single logout service.
    */
-  public slo(): Either<Error, IResponse<string>> {
-    return right({
-      body: "/",
-      status: 301
-    });
+  public async slo(): Promise<IResponsePermanentRedirect> {
+    const url: UrlFromString = {
+      href: "/"
```
Comment (cloudify) [inline src/controllers/authenticationController.ts]: @gunzip is this a valid `Url`?

File: src/services/messagesService.ts [hunk 4]
```diff
@@ -24,14 +31,22 @@ import { IApiClientFactoryInterface } from "./IApiClientFactory";
 
 const messageErrorOnUnknownResponse = "Unknown response.";
 const messageErrorOnApiError = "Api error.";
+const messageNotFound = "Not found.";
+
+export type messagesResponse<T> =
```
Comment (cloudify) [inline src/services/messagesService.ts]: type names must start with a capital letter

File: src/controllers/authenticationController.ts (hunk 3 above)
Comment (gunzip) [reply src/controllers/authenticationController.ts]: should use const errorOrUrl = UrlFromString.decode("/")

**Pull request merged**